llmfoundry/data/text_data.py

```diff
@@ -238,7 +235,7 @@ def build_text_dataloader(
                     cfg.dataset.get('split', None),
                     proportion=stream.get('proportion', None),
                     repeat=stream.get('repeat', None),
-                    samples=stream.get('samples', None),
```

Comment:
given the way we are accessing these args with `.get`, can we add some simple validation based on streaming version and the `stream` cfg dict? i know we have a full validation jira planned, but i think this one is likely to bite people.

Reply:
Yeah I can add some kind of validation, these `.get`s worry me too

Reply:
Would [**kwargs handling kwarg checking](https://github.com/mosaicml/llm-foundry/pull/292/commits/bd57a87a5ade3bc4e83fac02316ecf506040e540) do the trick?